### ISSUE TITLE
Migrate to Twitter Free API endpoints

### ISSUE THREAD

**Issue by Andronian42** [Andronian42/spdrFramePoster#8]
Still unsure about this one. Keep hearing reports that image upload is not possible, but the page advertising the free tier claims it's possible. Nonetheless, Free API should theoretically be able to post up to once every 30 minutes without hitting the limit by the end of the month, so it's definitely possible and fits this bot. Highly recommend doing this alongside #7 just in case. Obviously, this issue is time-sensitive (over the next 30 days? April 29th? thanks for making this so damned easy to plan, elon)

**Comment by Andronian42** [comment]
I also propose adding a note saying something along the lines of "fuck you Elon Musk" as a comment in the script

**Comment by Andronian42** [comment]
Looking over the API v2 endpoints, seems like it's similar to v1 in that media must be uploaded to the site separately before being attached to the tweet. Unlike v1, I cannot find a way to actually do this. This site from some AI company claims that 1.1 media upload endpoints will still be available, but I can't find an official source on that since the Twitter API pages sure as hell don't say anything about that. This is such a fucking mess lmao. Guess I'll implement v1.1 media upload alongside v2 posting and pray like hell.

**Comment by Andronian42** [comment]
I've done all I can for this on the social_revamp branch. Here's the situation I've found myself in:

According to what's been posted by Twitter Dev, the only API endpoints available for free users will be `POST /2/tweets` and `DELETE /2/tweets/:id`. According to one or two other sources I've found, `POST /1.1/media/upload` will be available as well.

As of now, this bot requires three endpoints to function as it does: `POST /1.1/media/upload` (for uploading images), `POST /1.1/media/metadata/create` (for adding alt text to images), and `POST /2/tweets` (for making tweets with the previously uploaded images). Locking out any of these three API functions would be problematic for the bot. If what I've read unofficially is wrong and the media upload endpoint will no longer be accessible to free users, needless to say, the bot will be dead.

So, bot status: as per usual, under Elon's Twitter, we're gonna have to sit and wait. I'll push these changes once I finish getting one or two other sites set up, then I'll set it all up on my server, and then we'll wait and see what happens with the API changes. We can only hope for the best, I suppose, but this really does suck for everybody. I hate this, and I hope you do too.
